### PR TITLE
RSDK-1878 - add MPU6050 tests (part 1)

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

**Review by kim-mishra — APPROVED** [review]
LGTM just had one question. Thanks for adding this!!

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		test.That(t, deps, test.ShouldBeEmpty)
+	})
+
+	t.Run("fails with no I2C bus", func(t *testing.T) {
+		cfg := AttrConfig{
+			BoardName: "thing",
+		}
+		_, err := cfg.Validate("path")
+		expectedErr := utils.NewConfigValidationFieldRequiredError("path", "i2c_bus")
+		test.That(t, err, test.ShouldBeError, expectedErr)
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go:37]: yeah, good point

**Review by penguinland — APPROVED** [review]
I don't understand the last test, but all my comments are tiny nits that you are welcome to ignore. LGTM!

Thank you for showing me an example of using dependency injection like this; I better understand how to mock out hardware now, which I have previously shied away from.

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+func TestValidateConfig(t *testing.T) {
+	t.Run("fails with no board supplied", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go:21]: I don't think there's any good reason to, I was previously re-using variables defined on the outside and when I refactored I missed this

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+			UseAlternateI2CAddress: true,
+		},
+	}
+	i2cHandle := &inject.I2CHandle{}
+	i2cHandle.ReadBlockDataFunc = func(ctx context.Context, b byte, u uint8) ([]byte, error) {
+		return []byte{0x68}, nil
+	}
+	wasClosed := false
+	i2cHandle.WriteByteDataFunc = func(ctx context.Context, b1, b2 byte) error {
+		if b2 != 0 {
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go:153]: The most significant thing that it's testing is that using `TryClose` actually calls the `Close` function on the sensor. Before my change to the `Close` function, this test was not passing. I'll change the variable name accordingly

File: components/movementsensor/mpu6050/mpu6050_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
+package mpu6050
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go:1]: Actually, this requires me making a lot of things public that shouldn't be. I should mention that Go's build process automatically ignores files with the pattern `"*_test.go"`, so there's no real concern here

[the remark this answers is not in the document]
Comment (penguinland) [reply components/movementsensor/mpu6050/mpu6050_test.go:1]: ohh, that's good to know! Thanks.

File: components/movementsensor/mpu6050/mpu6050_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (penguinland) [reply components/movementsensor/mpu6050/mpu6050_test.go:153]: Run that past me again. The change to the close function that I see is that you're now returning errors if they occur. How does that relate to this test?

[the remark this answers is not in the document]
Comment (gvaradarajan) [reply components/movementsensor/mpu6050/mpu6050_test.go]: components are closed by a call to the `TryClose` function defined in `utils`. This function silently fails, so we want to test to ensure that a component closes properly. TryClose requires that the component defines Close in a couple of accepted method signatures, neither of which was the signature of the MPU6050's previously implemented Close function.

[the remark this answers is not in the document]
Comment (penguinland) [reply components/movementsensor/mpu6050/mpu6050_test.go]: ...oh, I missed the `!` on this line! We're marking it as closed if we're writing any _nonzero_ value to any register. Now I get how this works. Thank you for finding and fixing the bug! I think the test would be a little more obvious if it was explicitly checking for us to write `1<<6` to register 107, but at least I get it.

**Pull request merged**